### PR TITLE
Xshard max, split, and select on Ray

### PR DESCRIPTION
**Pull request by jenniew**
Add min/max operation. Related issue: https://github.com/analytics-zoo/orca/issues/6 
Add train_test_split operation. Related issue: https://github.com/analytics-zoo/orca/issues/4
Add selection with column name/slice range.  Related issue: https://github.com/analytics-zoo/orca/issues/5
Add **kwargs support for read_csv, read_json. Related issue:https://github.com/analytics-zoo/orca/issues/20
Change DataShards.apply to DataShards.transform_shard. 
Add unit tests for all new APIs.
Fix xshard spark issues.
Refactor code.

### CODE REVIEW

File: pyzoo/test/zoo/xshard/test_ray_pandas.py [hunk 1]
```diff
@@ -74,18 +74,113 @@ def test_repartition(self):
 
     def test_apply(self):
```
Comment (jason-dai) [inline pyzoo/test/zoo/xshard/test_ray_pandas.py:75]: test_transform_shard

File: pyzoo/zoo/xshard/pandas/preprocessing.py [hunk 2]
```diff
@@ -239,27 +194,172 @@ def read_file_partitions(self, paths, file_type):
...
+                    df = pd.read_csv(path, **kwargs)
                 else:
                     raise Exception("Unsupported file type")
                 df_list.append(df)
         self.data = pd.concat(df_list)
         return 0
 
-    def apply(self, func, *args):
-        self.data = func(self.data, *args)
+    def transform_shard(self, func, *args, **kwargs):
```
Comment (jason-dai) [inline pyzoo/zoo/xshard/pandas/preprocessing.py:215]: change to `transform`

File: pyzoo/zoo/xshard/pandas/preprocessing.py [hunk 3]
```diff
@@ -239,27 +194,172 @@ def read_file_partitions(self, paths, file_type):
...
+            return self.data.loc[rows, columns]
+
+
+class PandasShard(object):
+    def __init__(self, shard, rows=None, columns=None):
+        self.ray_shard = shard
+        self.rows = rows
+        self.columns = columns
+
+    def transform_shard(self, func, *args, **kwargs):
```
Comment (jason-dai) [inline pyzoo/zoo/xshard/pandas/preprocessing.py:279]: change to `transform`

**Comment by jason-dai** [comment]
There are too many changes in this PR, which make it difficult to review and verify. We should make sure each PR only contains the minimum feature sets. One fundamental issue with the changes is that how to maintain multiple DataShards with one common set of Actors. For instance,
```
s = xshard.pands.read_csv(...)
s1 = s['user_id']
s1.tranform_shard(func) #what is the expected input of func? will it also change s?
```
And
```
s = xshard.pands.read_csv(...)
s1, s2 = train_test_split(s)
s.transform_shard(func) #will it change s1 and s2?
```
I suggest we close this PR and open several smaller ones (e.g., fixing https://github.com/analytics-zoo/orca/issues/20 first)

**Comment by jenniew** [comment]
> There are too many changes in this PR, which makes it difficult to review and verify. We should make sure each PR only contains the minimum feature sets. One fundamental issue with the changes is that how to maintain multiple DataShards with one common set of underlying Actors. For instance,
> 
> ```
> s = xshard.pands.read_csv(...)
> s1 = s['user_id']
> s1.tranform_shard(func) #what is the expected input of func? will it also change s?
> ```
> 
> And
> 
> ```
> s = xshard.pands.read_csv(...)
> s1, s2 = train_test_split(s)
> s.transform_shard(func) #will it change s1 and s2?
> ```
> 
> I suggest we close this PR and open several smaller ones (e.g., fixing [analytics-zoo/orca#20](https://github.com/analytics-zoo/orca/issues/20) first)

The current transform would change data in s. We need to change the implementation to return  new DataFrame or Series. Change Actor implementation or just use object id no Actor.
Already create PR: https://github.com/intel-analytics/analytics-zoo/pull/2305 for (https://github.com/analytics-zoo/orca/issues/20).
Close this PR.

**Pull request closed without merging**